### ISSUE TITLE
Backward compatibility for default resource parameters parsing

### ISSUE THREAD

**Issue by HiznyakNikita** [restsharp/RestSharp#1298]
## Expected Behavior
We are facing backward compatibility issues in RestSharp.
We used a 106.6.6 version of RestSharp and than update it to 106.6.9 and faced with default resource parameters parsing.
## Actual Behavior
We have code which communicates with third party API using requests for example like: 
"/v3/company/<companyId>/query?query=select * from Customer Where Metadata.LastUpdatedTime = '2015-03-01'&minorversion=4"
And create RestRequest like: 
```
var route = "/v3/company/<companyId>/query?query=select * from Customer Where Metadata.LastUpdatedTime = '2015-03-01'&minorversion=4";
var request = new RestRequest(route, Method.GET);
```
And get empty parameter list for request.
## Steps to Reproduce the Problem
  1. Create new RestRequest with:
```
var route = "/v3/company/<companyId>/query?query=select * from Customer Where Metadata.LastUpdatedTime = '2015-03-01'&minorversion=4";
var request = new RestRequest(route, Method.GET);
```
  2. Get empty parameters list

## Specifications
  - Version: from 106.6.7

##Reason
Before 106.6.7 version resource parameter in RestRequest parsed like: 
`System.Web.HttpUtility.ParseQueryString(Resource.Substring(queryStringStart));`
and after parsing we got a collection with two keys: 
1.query = select * from Customer Where Metadata.LastUpdatedTime = '2015-03-01'
2.minorversion = 4

But in eb5374c5e5ba5136b323b7dd65c7cb052e9dffd8 (Making it possible to have multiple parameters with the same name) commit pasing was changed to custom method:
`ParseQuery(Resource.Substring(queryStringStart + 1));`
With realization:
```
IEnumerable<NameValuePair> ParseQuery(string query) =>
 query.Split('&').Select(x =>
 {
 var pair = x.Split('=');
 return pair.Length == 2 ? new NameValuePair(pair[0], pair[1]) : NameValuePair.Empty;
 }).Where(x => !x.IsEmpty);
```
Where because of  "equality" signs count check the "query" parameter ignored, because it have another one "equality" sign inside: select * from Customer Where Metadata.LastUpdatedTime = '2015-03-01'
So this changes breaks possibility to have a query parameter with "equality" signs inside.

As possible solution we can remove parameters like this from resource string and add it later directly to request.Parameters collection. But i think because of backward compatibility the check
`pair.Length == 2`
must be removed inside ParseQuery method to avoid this problem

**Comment by alexeyzimarev** [comment]
Please consider using `request.AddQueryParameter` method instead of putting the query to the request URL. In this case, the query parameter will be properly encoded and you won't get this issue. That is actually the correct usage of RestSharp.

**Comment by HiznyakNikita** [comment]
thanks!
you are right because in [RFC 3986](https://tools.ietf.org/html/rfc3986#section-2.2) indicated that equal sign is reserved character.
sure, i can use AddQueryParameter as i already done.
but it will be great do changes like this using some Obsolete labels, because some API don't developed in accordance with all conventions, and changes like this breaks functionality totally (as in our case)

**Comment by alexeyzimarev** [comment]
I agree. At the same time, I am not sure where could I've put the obsolete label?

**Comment by HiznyakNikita** [comment]
yes, in this case it was hard because parsing logic was directly inside constructor
maybe if logic was at separate method (something like parse query parameters) it was easier

**Comment by alexeyzimarev** [comment]
I would be happy to get a PR to fix this if it is a real issue.

**Comment by alexeyzimarev** [comment]
It is fixed in 106.9.
